tools/walletextension/multi_acc_helper.go

```diff
@@ -46,7 +46,7 @@ func suggestAccountClient(req *rpcRequest, accClients map[common.Address]*rpc.En
 	if req.method == rpc.RPCCall {
 		// Otherwise, we search the `data` field for an address matching a registered viewing key.
 		addr, err := searchDataFieldForAccount(paramsMap, accClients)
-		if err == nil {
+		if err == nil && addr != nil {
```

Comment:
I don't think this is quite the right solution, instead `searchDataFieldForAccount` should never return `nil,nil` because that breaks the Go convention. If there was no Account then it's an error imo.

I would change line 92 to return an error instead of nil,nil

Reply:
Yes, I have a pet peeve with Not Found being an error.  But I don't think it applies here.

Good suggestion, done 👍